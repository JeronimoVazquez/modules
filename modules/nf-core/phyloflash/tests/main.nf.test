nextflow_process {

    name "Test Process PHYLOFLASH"

    script "../main.nf"
    process "PHYLOFLASH"

    tag "modules"
    tag "modules_nfcore"
    tag "phyloflash"

    // only stub runs as a database has to be downloaded
    test("test_phyloflash_single_end -- stub") {
        options '-stub'
        when {
            params {
                module_args = ''
            }
            process {
                """
                input[0] = [
                    [id:'test', single_end: true],
                    [file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists:true)]
                    ]
                input[1] = []
                input[2] = []
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}