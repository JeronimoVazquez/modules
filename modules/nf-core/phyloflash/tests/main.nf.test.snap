{
    "test_phyloflash_single_end -- stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        [
                            "test.SSU.collection.fasta:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test.phyloFlash:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,f508dc3d229d5278006a976e32784a84"
                ],
                "results": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        [
                            "test.SSU.collection.fasta:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test.phyloFlash:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f508dc3d229d5278006a976e32784a84"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-13T15:22:36.509231028"
    }
}